plugins/autostandby.koplugin/main.lua

```diff
@@ -0,0 +1,166 @@
+local Device = require("device")
+
+if not Device:isPocketBook() --[[and not Device:isKobo()]] then
+    return { disabled = true, }
+end
+
+local PowerD = Device:getPowerDevice()
+local DataStorage = require("datastorage")
+local LuaSettings = require("luasettings")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local SpinWidget = require("ui/widget/spinwidget")
+local logger = require("logger")
+local _ = require("gettext")
+
+local AutoStandby = WidgetContainer:new{
+    is_doc_only = false,
+    name = "autostandby",
+    settings = LuaSettings:open(DataStorage:getSettingsDir() .. "/autostandby.lua"),
+    delay = 0,
+    lastInput = 0,
+    preventing = false,
+}
+
+function AutoStandby:init()
+    if not self.settings:has('filter') then
+        logger.dbg("AutoStandby: No settings found, initializing defaults")
+        self.settings
+        -- If forbidden, standby is never allowed to occur
+        :saveSetting('forbidden', false)
+        -- Consider input only further than this many seconds apart
+        :saveSetting('filter', 1)
```

Comment:
Not used to see that kind of constructs in our code base :) but I guess it's fine - the alternative being assigning a plain table with these settings directly to self.settings.data I think.

Reply:
Yeah, I'm not sure what this actually does, actually (lua is weird)? implied self?

Reply:
Not Lua, but may be our Luasettings :) It's just that most of our `LuaSettings:saveSetting/delSetting/flipNilOrFalse` return self, so you can chain them like that.
I think that's something quite common in Javascript, to chain stuff like that.

Reply:
We normally prefer double quotes btw.

Reply:
Switched to setting data directly. This introduces new contract (of data being settable), but I reckon there's no scenario where that would hurt, as all the funky MT stuff should be hung on top of the settings object.